include/net/nrf_cloud_rest.h

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file nrf_cloud_rest.h
+ *
+ * @brief nRF Cloud REST API.
+ */
+#ifndef NRF_CLOUD_REST_H__
+#define NRF_CLOUD_REST_H__
+
+#include <zephyr/types.h>
+#include <net/nrf_cloud.h>
+#include <net/nrf_cloud_pgps.h>
+#include <net/nrf_cloud_cell_pos.h>
+#include <modem/lte_lc.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** @defgroup nrf_cloud_rest nRF Cloud REST API
+ * @{
+ */
+
+/** @brief HTTP status codes returned from nRF Cloud */
+enum nrf_cloud_http_status {
+	NRF_CLOUD_HTTP_STATUS_UNHANDLED = -1,
+	NRF_CLOUD_HTTP_STATUS_NONE = 0,
+	NRF_CLOUD_HTTP_STATUS_OK = 200,
+	NRF_CLOUD_HTTP_STATUS_ACCEPTED = 202,
+	NRF_CLOUD_HTTP_STATUS_PARTIAL = 206,
+	NRF_CLOUD_HTTP_STATUS_BAD_REQ = 400,
+	NRF_CLOUD_HTTP_STATUS_UNAUTH = 401,
+	NRF_CLOUD_HTTP_STATUS_FORBIDDEN = 403,
+	NRF_CLOUD_HTTP_STATUS_NOT_FOUND = 404,
+	NRF_CLOUD_HTTP_STATUS_BAD_RANGE = 416,
+	NRF_CLOUD_HTTP_STATUS_UNPROC_ENTITY = 422,
+};
+
+/** @brief nRF Cloud AGPS REST request types */
+enum nrf_cloud_rest_agps_req_type {
+	NRF_CLOUD_REST_AGPS_REQ_ASSISTANCE,
+	NRF_CLOUD_REST_AGPS_REQ_LOCATION,
+	NRF_CLOUD_REST_AGPS_REQ_CUSTOM,
+	NRF_CLOUD_REST_AGPS_REQ_UNSPECIFIED,
+};
+
+/** @brief Parameters and data for using the nRF Cloud REST API */
+struct nrf_cloud_rest_context {
+
+	/** Connection socket; initialize to -1 and library
+	 * will make the connection.
+	 */
+	int connect_socket;
+	/** If the connection should remain after API call */
+	bool keep_alive;
+	/** Timeout value for receiving response data */
+	int32_t timeout_ms;
+	/** Authentication string: JWT @ref modem_jwt */
+	char *auth;
+	/** User allocated buffer for receiving API response.
+	 * Buffer size should be limited according to the
+	 * maximum TLS receive size of the modem.
+	 */
+	char *rx_buf;
+	/** Size of rx_buf */
+	size_t rx_buf_len;
+	/** Fragment size for downloads, set to zero to use
+	 * :option:`CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE`.
```

Comment:
You still have this :option: directive here. Please leave it out. It does not work in doxygen.